src/jupynium/lua/helpers.lua

```diff
@@ -688,3 +688,36 @@ function Jupynium_kernel_complete_async(bufnr, code_line, col, callback)
 
   Jupynium_rpcnotify("kernel_complete_async", bufnr, true, code_line, col, callback_id)
 end
+
+function Jupynium_connect_kernel(bufnr, hostname)
+  if bufnr == nil or bufnr == 0 then
+    bufnr = vim.api.nvim_get_current_buf()
+  end
+  if Jupynium_syncing_bufs[bufnr] == nil then
+    Jupynium_notify.error { [[Cannot get kernel list without synchronising.]], [[Run `:JupyniumStartSync`]] }
+    return
+  end
+  local kernel_id = Jupynium_rpcrequest("kernel_connect_info", bufnr, true)
+  local cmd = "jupyter console --existing " .. kernel_id
```

Comment:
You can use `jupyter_command` from options. You should do `pcall require ...` to check if Jupynium installation exists and use that, if not installed or can't find the option, just use `"jupyter"`

Reply:
I'm not quite sure what you mean here. Are you saying to use `pcall` to check if `jupyter-console` is executable?

Reply:
Jupynium should run without installing the plugin. The only thing needed is the python command to use Jupynium. The plugin is just a wrapper around the python command to call jupynium easier.

This design is very important as Jupynium needs to attach multiple instances of neovim. We don't know if each one has different versions of the plugin installed, and that shouldn't affect anything. Sometimes it's even connecting to remote neovim, and I didn't want to have a requirement that the version of the vim plugin and the python command must match. It's a very special design that is different from other plugins.

So, what you need to do is to make sure your patch works with `nvim --clean --listen /tmp/jupynium-nvim` and `jupynium --nvim_listen_addr /tmp/jupynium-nvim`

Reply:
Did you mean change it to
```lua
  local jupyter_command = "jupyter"
  local ok, options = pcall(require, "jupynium.options")
  if ok then
    jupyter_command = options.opts.jupyter_command
  end
  local cmd = jupyter_command .. " console --existing " .. kernel_id
```
I still don't quite understand how Jupyter Notebook works, what else do I need to do to ensure that it can work or is that enough?

Reply:
I play a bit attach remote neovim, that is really amazing experience. So we use `jupynium.options` if jupynium is installed and fallback to defaults command if not?

Reply:
something like that, yes. But `jupyter_command` can also be a list-like table.

Reply:
How about that look like now? I'm not sure how to deal with conversation, just mark it resolved?